build broken with chip_enable_wifi=false

#### Problem
Build breakage as follows:

```
onnectedhomeip$ uname -a 
Linux ubuntu 5.4.0-58-generic #64-Ubuntu SMP Wed Dec 9 08:16:25 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
msandstedt@ubuntu:~/hub-core/lib/connectedhomeip$ git log -n1
commit f60ffc5760c1bbc018d750e36a718bd199dc4b4f (HEAD -> master, upstream/master, origin/master, origin/HEAD)
Author: Song Guo <songguo@google.com>
Date:   Tue Dec 15 14:20:31 2020 +0800

    [doc] Add document for chip device controller python cli (#4169)
    
    * [doc] Add document for chip device controller python cli
    
    * Add note about building bluez
    
    * Update doc
connectedhomeip$ . scripts/activate.sh 

  WELCOME TO...

  ▄███▒  ░▓█  ░▓█ ░▓█▓ ▒█████▄
 ██▒ ▀█▒  ▒█   ▒█  ░█▒  ▒█░  █░
 █▓░      ▒██████  ░█▒  ▒█▄▄▄█░
 ▓█   █▒  ▒█   ▒█  ░█░  ▒█▀
 ░▓███▀  ░▓███░▓█▒ ░█░  ▒█

  ACTIVATOR! This sets your shell environment variables.

Activating environment (setting environment variables):

  Setting environment variables for CIPD package manager...done
  Setting environment variables for Python environment.....done
  Setting environment variables for Host tools.............done

Sanity checking the environment:

20201215 12:51:29 INF Environment passes all checks!

Environment looks good, you are ready to go!

connectedhomeip$ gn gen out --args="chip_enable_wifi=false"
Done. Made 348 targets from 110 files in 169ms
connectedhomeip$ ninja -C out
ninja: Entering directory `out'
[1/112] c++ obj/examples/shell/shell_common/shell_common.cmd_device.cpp.o
FAILED: obj/examples/shell/shell_common/shell_common.cmd_device.cpp.o 
g++ -MMD -MF obj/examples/shell/shell_common/shell_common.cmd_device.cpp.o.d -O0 -g2 -fno-common -ffunction-sections -fdata-sections -fno-exceptions -fPIC -Wall -Wextra -Wshadow -Werror -Wno-deprecated-declarations -Wno-unknown-warning-option -Wno-missing-field-initializers -Wno-unused-parameter -Wno-psabi -Wno-cast-function-type -fdiagnostics-color -fno-strict-aliasing -D_REENTRANT -isystem../third_party/ot-br-posix/repo/src -isystem../third_party/ot-br-posix/repo/include -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -I/usr/include/gio-unix-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -std=gnu++14 -fno-rtti -Wnon-virtual-dtor -DENABLE_CHIP_SHELL -DOPENTHREAD_CONFIG_CLI_TRANSPORT=OT_CLI_TRANSPORT_CONSOLE -DCHIP_HAVE_CONFIG_H=1 -DCHIP_DEVICE_CONFIG_ENABLE_CHIPOBLE=true -DLWIP_IPV4=1 -DLWIP_IPV6=1 -DLWIP_API=1 -DLWIP_ETHERNET=1 -DLWIP_SLIP=0 -DLWIP_6LOWPAN=1 -DLWIP_PPP=1 -I../examples/shell/shell_common -I../examples/shell/shell_common/include -I../src/lib/shell -I../src/app/gen -I../src/app/util -I../src/app -Igen/src/app/include -I../src/include -I../src -Igen/include -I../src/lib -I../config/standalone -I../third_party/nlassert/repo/include -I../third_party/nlio/repo/include -I../third_party/nlfaultinjection/repo/include -I../src/lwip/standalone -I../third_party/lwip/repo/lwip/src/include -I../src/lwip/include -I../src/setup_payload -Igen -I../third_party/inipp/repo/inipp -c ../examples/shell/shell_common/cmd_device.cpp -o obj/examples/shell/shell_common/shell_common.cmd_device.cpp.o
In file included from ../src/include/platform/ConnectivityManager.h:269,
                 from ../src/include/platform/CHIPDeviceLayer.h:29,
                 from ../examples/shell/shell_common/cmd_device.cpp:28:
../src/platform/Linux/ConnectivityManagerImpl.h:181:40: error: no declaration matches ‘chip::DeviceLayer::ConnectivityManager::WiFiAPMode chip::DeviceLayer::ConnectivityManagerImpl::_GetWiFiAPMode()’
  181 | inline ConnectivityManager::WiFiAPMode ConnectivityManagerImpl::_GetWiFiAPMode()
      |                                        ^~~~~~~~~~~~~~~~~~~~~~~
```

 #### Proposed Solution
```
diff --git a/src/platform/Linux/ConnectivityManagerImpl.h b/src/platform/Linux/ConnectivityManagerImpl.h
index f2298d3e..fbcd9e05 100644
--- a/src/platform/Linux/ConnectivityManagerImpl.h
+++ b/src/platform/Linux/ConnectivityManagerImpl.h
@@ -178,6 +178,7 @@ private:
     uint32_t mWiFiAPIdleTimeoutMS;
 };
 
+#if CHIP_DEVICE_CONFIG_ENABLE_WPA
 inline ConnectivityManager::WiFiAPMode ConnectivityManagerImpl::_GetWiFiAPMode()
 {
     return mWiFiAPMode;
@@ -202,6 +203,7 @@ inline bool ConnectivityManagerImpl::_HaveServiceConnectivity()
 {
     return _HaveServiceConnectivityViaThread();
 }
+#endif
 
 /**
  * Returns the public interface of the ConnectivityManager singleton object.
```

--- a/src/app/MessageDef.cpp
+++ b/src/app/MessageDef.cpp
@@ -142,6 +142,7 @@ static void PrettyPrintIM(bool aIsNewLine, const char * aFmt, ...)
 #define PRETTY_PRINT_CHECKPOINT()
 #define PRETTY_PRINT(fmt, ...)
 #define PRETTY_PRINT(fmt, ...)
+#define PRETTY_PRINT_SAMELINE(fmt, ...)
 #define PRETTY_PRINT_INCDEPTH()
 #define PRETTY_PRINT_DECDEPTH()
 #endif // CHIP_DETAIL_LOGGING

--- a/src/platform/Linux/ConnectivityManagerImpl.h
+++ b/src/platform/Linux/ConnectivityManagerImpl.h
@@ -178,6 +178,7 @@ private:
     uint32_t mWiFiAPIdleTimeoutMS;
 };
 
+#if CHIP_DEVICE_CONFIG_ENABLE_WPA
 inline ConnectivityManager::WiFiAPMode ConnectivityManagerImpl::_GetWiFiAPMode()
 {
     return mWiFiAPMode;
@@ -202,6 +203,7 @@ inline bool ConnectivityManagerImpl::_HaveServiceConnectivity()
 {
     return _HaveServiceConnectivityViaThread();
 }
+#endif
 
 /**
  * Returns the public interface of the ConnectivityManager singleton object.

